ui/app/selectors/transactions.js

```diff
@@ -251,11 +256,26 @@ export const nonceSortedTransactionsSelector = createSelector(
           nonceProps.hasCancelled = true
         }
       } else {
+        /*
+          Error:
+          {
+            message: "Error: [ethjs-rpc] rpc error with payload {\"id\":428016472614,\"jsonrpc\":\"2.0\",\"params\":[\"0xf8650a843b9aca00843b9aca00940bd88c9c99be119dbbcd493699bfbffab0904d6980802ca0f511cd1297503b5a670e9459a174eceb4a2b7786df212821873eb2302727a49ba05ac22d6732af3e21539a2cbb738f53ccac2477de1e5a2d2a389f46b88693432f\"],\"method\":\"eth_sendRawTransaction\"} [object Object]",
+​​​            rpc: {
+              code: -32000
+              message: "exceeds block gas limit"
+            }
+          }
+        */
+        let primaryTransaction = transaction
+        if(transaction.status === FAILED_STATUS && Boolean(transaction?.err?.message)) {
```

Comment:
David pointed out something on a call we had where he went over some of his findings. the status 'failed' is not in the PRIORITY_STATUS_HASH.

The only way for a failed transaction to be the primary transaction is simply it being the latest transaction for a nonce. The `transactionSelector` that is a predicate of this selector, returns a list of transactions sorted by time. Therefore cancellations, retries, and transactions submitted with custom nonce equal to an existing transaction nonce will always be the `primaryTransaction` for a group.

`initialTransaction` gets set appropriately due to the comparison of time

I don't know if only network failures have this error object, but if that's true then that seems like a pretty good way to draw the distinction.